tests/test_trusted_metadata_set.py

```diff
@@ -274,9 +274,24 @@ def version_modifier(timestamp: Timestamp) -> None:
         with self.assertRaises(exceptions.BadVersionNumberError):
             self.trusted_set.update_timestamp(self.metadata[Timestamp.type])
 
+    def test_update_timestamp_with_same_timestamp(self) -> None:
+        # Test that timestamp is NOT updated if:
+        # new_timestamp.version == trusted_timestamp.version
+        self.trusted_set.update_timestamp(self.metadata[Timestamp.type])
+        initial_timestamp = self.trusted_set.timestamp
+
+        # Update timestamp with the same version.
+        self.trusted_set.update_timestamp((self.metadata[Timestamp.type]))
+
+        # Every object has a unique id() if they are equal, this means timestamp
+        # was not updated.
+        self.assertEqual(id(initial_timestamp), id(self.trusted_set.timestamp))
```

Comment:
This works because we know that `update_timestamp` assigns a new object to `self.trusted_set.timestamp` in case of an update. But the test wouldn't detect a change in the same object. E.g.

```python
class Timestamp: pass

# Create initial object
timestamp = Timestamp()

# Copy reference
initial_timestamp = timestamp

# Actually update initial object
timestamp.attr_change = "not detected"
 
# Change not detected
assert id(initial_timestamp) == id(timestamp)
```

Reply:
IMO it's safer to make a deep copy and assertEqual the objects. 🤷

Reply:
I just realized that if we do use `deepcopy()` we are creating a new object for `initial_timestamp` whose id is not equal to the original `trusted_set.timestamp` id.
This means that even if our code works perfectly fine and doesn't update the `trusted_set.timestamp` when the new_timestamp has the same version then the assert `self.assertEqual(id(initial_timestamp), id(self.trusted_set.timestamp))` will not be true.

We need exactly a reference to an object which we create with a normal assignment because if the `initial_timestamp` reference points to the same object as the `trusted_set.timestamp` reference this means that an update was not finished. 

I think it should be safe to use the normal assignments as we want to verify not that the content of the initial_timestamp is not changed, but that it's actually the same object as the `trusted_set.timestamp` after the update.

Reply:
> we want to verify not that the content of the initial_timestamp is not changed, but that it's actually the same object as the trusted_set.timestamp after the update.

Okay, in that case it makes sense to use `id`.